clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/ingestion/heartbeat/HeartbeatVersionedStats.java

```diff
@@ -23,4 +32,32 @@ public void recordLeaderLag(String storeName, int version, String region, long l
   public void recordFollowerLag(String storeName, int version, String region, long lag) {
     getStats(storeName, version).recordFollowerLag(region, lag);
   }
+
+  @Override
+  public synchronized void loadAllStats() {
+    // No-op
+  }
+
+  @Override
+  public void handleStoreCreated(Store store) {
+    if (isStoreAssignedToThisNode(store.getName())) {
```

Comment:
When would this get called? The store will be created before a version is created. After a version is created, a SN will get assigned a partition. So, do we need this? Can we modify `handleStoreChanged` to call `addStore` only when `isStoreAssignedToThisNode` is `true`?

Reply:
Overriding addStore might actually be the better way to go as I think about it.

Reply:
oof.  Actually I remember why I didn't do this for addStore.  addStore isn't a void method and I was worried about triggering NPE's somewhere else.  I guess coming back to storeCreated.  storeCreated should honestly probably NoOp as well as you are correct.  There's no way a version will be created when a store is first created, so theres not really any reason to react here.